sys/dev/nvme/nvme.c

```diff
@@ -310,18 +314,18 @@ nvme_register_consumer(nvme_cons_ns_fn_t ns_fn, nvme_cons_ctrlr_fn_t ctrlr_fn,
 {
 	int i;
 
-	/*
-	 * TODO: add locking around consumer registration.
-	 */
 	for (i = 0; i < NVME_MAX_CONSUMERS; i++)
 		if (nvme_consumer[i].id == INVALID_CONSUMER_ID) {
+			mtx_lock(&nvme_consumer_mutex);
```

Comment:
This is wrong.

Assume two threads concurrently register consumer, they may still ref to the same nvme_consumer[i] , then this lock does not serve any purpose.